mlir/lib/Conversion/MemRefToSPIRV/MemRefToSPIRV.cpp

```diff
@@ -445,15 +445,23 @@ DeallocOpPattern::matchAndRewrite(memref::DeallocOp operation,
 // LoadOp
 //===----------------------------------------------------------------------===//
 
-using AlignmentRequirements =
+using MemoryRequirements =
     FailureOr<std::pair<spirv::MemoryAccessAttr, IntegerAttr>>;
```

Comment:
Now that it's less clear what this represents (e.g., what is the interpretation of `IntegerAttr` when there's no alignment), I think we should make it a struct and add some comments.
Also the `FailureOr` should probably be spelled out as the return type and not part of the typedef.